src/openloco/objects/vehicle_object.h

```diff
@@ -4,6 +4,25 @@
 
 namespace openloco
 {
+    enum class TransportMode : uint8_t
+    {
+        rail = 0,
+        road,
+        air,
+        water
+    };
+
+    enum class VehicleType : uint8_t
+    {
+        train = 0,
+        bus,
+        truck,
+        tram,
+        plane,
+        ship,
+        Count
```

Comment:
Is adding `count` a good idea? It'll mean that switch statements will automatically have a `case VehicleType::count`: added. I'd rather see a `vehicleTypeCount` constant, or a list that can be foreached over.

Reply:
I suggested this in the other PR, but concede it's not the best idea, in retrospect.

Reply:
I was just following what was done in openrct2. I can change it to a constexpr if that is preferred.